### PR TITLE
feat: Fan out project config fetches

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: server/src/actors/project.rs [hunk 1]
```diff
@@ -894,73 +903,94 @@ impl ProjectCache {
...
+                    .send(SendQuery(request))
+                    .map_err(ProjectError::ScheduleFailed)
+                    .map(move |response| (channels_batch, request_start, response))
+            })
+            .collect();
 
-        self.upstream
-            .send(SendQuery(request))
-            .map_err(ProjectError::ScheduleFailed)
+        future::join_all(requests)
```
Comment (jan-auer) [inline server/src/actors/project.rs:949]: Can you please add a comment about the behavior of `join_all` here? To double check: All requests run to completion (even with failures), unless the mailbox of the `Upstream` actor is full. If one request fails, the error is handled explicitly below.

File: server/src/actors/project.rs [hunk 2]
```diff
@@ -894,73 +903,94 @@ impl ProjectCache {
...
+
+                let request = GetProjectStates {
+                    projects: channels_batch.keys().copied().collect(),
+                    #[cfg(feature = "processing")]
+                    full_config: self.config.processing_enabled(),
+                };
+
+                // count number of http requests for project states
+                metric!(counter("project_state.request") += 1);
+                let request_start = Instant::now();
```
Comment (jan-auer) [inline server/src/actors/project.rs]: To reduce the size of the tuple below, how about we move this out of this closure and share it between all batches?

File: server/src/actors/project.rs [hunk 3]
```diff
@@ -894,73 +903,94 @@ impl ProjectCache {
...
+
+                self.upstream
+                    .send(SendQuery(request))
+                    .map_err(ProjectError::ScheduleFailed)
+                    .map(move |response| (channels_batch, request_start, response))
+            })
+            .collect();
```
Comment (jan-auer) [inline server/src/actors/project.rs:945]: IIRC, `join_all` just requires an iterator, so you can omit the `collect` and just pass it directly.

**Review by jan-auer — COMMENTED** [review]
Looks good on a high level, but I'd like to do a final walk-through with you in person before we merge this.
A few nit picks below.

File: server/src/actors/project.rs (hunk 3 above)
Comment (untitaker) [reply server/src/actors/project.rs:945]: doesn't work due to lifetime issues

File: server/src/actors/project.rs (hunk 1 above)
Comment (untitaker) [reply server/src/actors/project.rs:949]: done

File: server/src/actors/project.rs (hunk 2 above)
Comment (untitaker) [reply server/src/actors/project.rs]: done

**Comment by untitaker** [comment]
@jan-auer I tried your suggestion of using only a single channels map. The issue is that we need to know which channels belong to a request if the request fails, so that we can put them back into `self.state_channels`. We could chunk the vector of IDs and pass around those chunks, but that's still not getting rid of passing around something next to the response. I feel like it's not worth it.

**Review by jan-auer — APPROVED** [review]

**Pull request merged**